spec/unit/provider/exec/powershell_spec.rb

```diff
@@ -115,6 +116,19 @@
       }
     MANIFEST
     }
+    let(:tmpdir) { Dir.mktmpdir('statetmp').encode!(Encoding::UTF_8) }
+
+    before :each do
+      # a statedir setting must now exist per the new transactionstore code
+      # introduced in Puppet 4.6 for corrective changes, as a new YAML file
+      # called transactionstore.yaml will be written under this path
+      # which defaults to c:\dev\null when not set on Windows
+      Puppet[:statedir] = tmpdir
+    end
+
+    after :each do
+      FileUtils.rm_rf(tmpdir)
+    end
```

Comment:
Yeah, I could have moved it further out... but... there's also the matter of this only being applicable to a couple of tests.  And those tests are a bit outside the norm of our regular unit tests given they use Puppet as a library to compile manifests.